File: python/Siak.py
#import data dari file Human
from Human import Human
#Siak yang merupakan anak dari Human
class Siak(Human):
    
    #constructor
    def __init__(self, asalUniv = "", emailEdu = "", nik = 0, nama = "", jenisKelamin = ""):
        #untuk mengakses variabel dan metode pada parent
        super().__init__(nik, nama, jenisKelamin)
        self.asalUniv = asalUniv
        self.emailEdu = emailEdu
    
    #setter untuk asal universitas dan email edu
    def setAsalUniv(self, asalUniv):
        self.asalUniv = asalUniv
    
    def setEmailEdu(self, emailEdu):
        self.emailEdu = emailEdu
    
    #getter untuk asal universitas dan email edu
    def getAsalUniv(self):
        return self.asalUniv
    
    def getEmailEdu(self):
        return self.emailEdu

File: c++/Program/Siak.cpp
#include "Header.hh"

//construktor
Siak::Siak() : Human(){
    email_edu = "";
    asal_univ = "";
}

Siak::Siak(int nik, string nama, string jenis_kelamin){
}

Siak::Siak(int nik, string nama, string jenis_kelamin, string asal_univ, string email_edu) : Human (nik, nama, jenis_kelamin){
    this->email_edu = email_edu;
    this->asal_univ = asal_univ;
}

//getter and setter
void Siak::setAsalUniv(string asal_univ){
    this->asal_univ = asal_univ;
}
string Siak::getAsalUniv(){
    return this->asal_univ;
}

void Siak::setEmailEdu(string email_edu){
    this->email_edu = email_edu;
}
string Siak::getEmailEdu(){
    return this->email_edu;
}

//destructor
Siak::~Siak(){

}